controllers/configurationpolicy_controller.go

```diff
@@ -2711,10 +2741,58 @@ func (r *ConfigurationPolicyReconciler) checkAndUpdateResource(
 
 		updateSucceeded = true
 	} else {
+		if throwSpecViolation && recordDiff != policyv1.RecordDiffNone {
+			// The spec didn't require a change but throwSpecViolation indicates the status didn't match. Handle
+			// this diff for this case.
+			mergedObjCopy := obj.existingObj.DeepCopy()
+			removeFieldsForComparison(mergedObjCopy)
+
+			// The provided isInform value is always true because the status checking can only be inform.
+			diff = handleDiff(log, recordDiff, true, existingObjectCopy, mergedObjCopy)
+		}
+
 		r.setEvaluatedObject(obj.policy, obj.existingObj, !throwSpecViolation)
 	}
 
-	return throwSpecViolation, "", updateNeeded, updateSucceeded
+	return throwSpecViolation, "", diff, updateNeeded, updateSucceeded
+}
+
+// handleDiff will generate the diff and then log it or return it based on the input recordDiff value. If recordDiff
+// is set to None or is set to InStatus with enforce, no diff is generated. When recordDiff is set to Sensored, a
+// message indicating so is returned.
+func handleDiff(
+	log logr.Logger,
+	recordDiff policyv1.RecordDiff,
+	isInform bool,
+	existingObject *unstructured.Unstructured,
+	mergedObject *unstructured.Unstructured,
+) string {
+	if recordDiff == policyv1.RecordDiffNone {
+		return ""
+	}
+
+	if !isInform && recordDiff == policyv1.RecordDiffInStatus {
+		return ""
+	}
+
+	computedDiff, err := generateDiff(existingObject, mergedObject)
+	if err != nil {
+		log.Info("Failed to generate the diff: " + err.Error())
+
+		return ""
+	}
+
+	switch recordDiff { //nolint:exhaustive
```

Comment:
I would prefer to make the switch cases exhaustive, I think it makes things a tiny bit more clear.